### ISSUE TITLE
Errors after upgrading to io-ts 2.2.17

### ISSUE THREAD

**Issue by ehtb** [mmkal/ts#259]
After upgrading io-ts to 2.2.17, a definition as simple as:

```
t.type({
  adults: optional(t.union([t.number, t.null])),
})
or
sparseType({
  adults: optional(t.union([t.number, t.null])),
})
```

Shows the following error:

```
Argument of type 'UnionC<[NumberC, NullC]>' is not assignable to parameter of type 'Mixed'.
  Types of property 'pipe' are incompatible.
    Type '<B, IB, A extends IB, OB extends A>(this: Type<A, number | null, unknown>, ab: Type<B, OB, IB>, name?: string | undefined) => Type<B, number | null, unknown>' is not assignable to type '<B, IB, A extends IB, OB extends A>(this: Type<A, any, unknown>, ab: Type<B, OB, IB>, name?: string | undefined) => Type<B, any, unknown>'.
      The 'this' types of each signature are incompatible.
        Type 'Type<A, any, unknown>' is not assignable to type 'Type<IB, number | null, unknown>'.
          Types of property 'is' are incompatible.
            Type 'Is<A>' is not assignable to type 'Is<IB>'.
              Type predicate 'u is A' is not assignable to 'u is IB'.
                Type 'A' is not assignable to type 'IB'.
                  'IB' could be instantiated with an arbitrary type which could be unrelated to 'A'
```

Downgrading to io-ts 2.2.16 removes the error

**Comment by mmkal** [comment]
I'm afraid to say I abandoned io-ts in favour of zod a while ago, since a lot of the features in this library are built in (and zod seems geared towards human beings who want to write typescript, unlike io-ts which is geared towards [mathematicians who don't](https://twitter.com/giuliocanti/status/1235521019131170817). I used to be a mathematician myself, so no shade - I just am not one anymore.

Not all the features exist in zod, though, and I miss some of them. If you are still on the old version of io-ts and using this library, could you tell me which features you are using? I will likely split off the differentiating/useful features of io-ts-extra into a new library, so your feedback would be helpful!

TL;DR for this issue: I probably will not fix this.
